### PR TITLE
Power law creep combine ad/nonad

### PR DESCRIPTION
**Pull request by tophmatthews**
Combine Ad/nonAD in a model that also uses ChainedADReal/ChainedReal.
Move power law creep tests out of `combined` into `tensor_mechanics` where they really belong by trading out heat conduction objects for standard moose objects.

Ref #21009

### CODE REVIEW

**Comment by tophmatthews** [comment]
Requires #20997

File: modules/tensor_mechanics/src/materials/PowerLawCreepStressUpdate.C [hunk 1]
```diff
@@ -28,92 +30,111 @@ PowerLawCreepStressUpdate::validParams()
...
+template ADReal
+PowerLawCreepStressUpdateTempl<true>::computeResidualInternal<ADReal>(const ADReal &,
+                                                                      const ADReal &);
 template ChainedReal
-PowerLawCreepStressUpdate::computeResidualInternal<ChainedReal>(const Real &, const ChainedReal &);
+PowerLawCreepStressUpdateTempl<false>::computeResidualInternal<ChainedReal>(const Real &,
+                                                                            const ChainedReal &);
+template ChainedADReal
+PowerLawCreepStressUpdateTempl<true>::computeResidualInternal<ChainedADReal>(const ADReal &,
+                                                                             const ChainedADReal &);
```
Comment (tophmatthews) [inline modules/tensor_mechanics/src/materials/PowerLawCreepStressUpdate.C:140]: The main reason of this PR (beyond the obvious code cleanup) is to prove out this concept with the AD/nonAD and ChainedReal/Real templates. Does this section make sense to you @dschwen ?

Comment (dschwen) [reply modules/tensor_mechanics/src/materials/PowerLawCreepStressUpdate.C:140]: couldn't you simply replace that second layer of templating with `GenericChainedReal<is_ad>`?

Comment (dschwen) [reply modules/tensor_mechanics/src/materials/PowerLawCreepStressUpdate.C:140]: Oh, never mind. I didn't see those other combos

Comment (tophmatthews) [reply modules/tensor_mechanics/src/materials/PowerLawCreepStressUpdate.C:140]: Yeah...kinda guess in check here, so please see if this makes sense to you

**Review by dschwen — APPROVED** [review]
looks ok to me

**Pull request merged**